components/Code/index.jsx

```diff
@@ -38,16 +41,25 @@ CopyCode.propTypes = {
 };
 
 function Code(props) {
-  const { children, className, copyButtons, lang, meta, theme } = props;
+  const { children, className, copyButtons, lang, meta } = props;
 
   const langClass = className.search(/lang(?:uage)?-\w+/) >= 0 ? className.match(/\s?lang(?:uage)?-(\w+)/)[1] : '';
   const language = canonicalLanguage(lang) || langClass;
 
   const codeRef = React.createRef();
+
+  const colorMode = document.querySelector('[data-color-mode]').getAttribute('data-color-mode');
+  const initialTheme = localStorage.getItem('color-scheme');
+  const userColorScheme = useMediaQuery('(prefers-color-scheme: dark)') ? 'dark' : 'light';
+
+  const colorScheme = colorMode === 'auto' && initialTheme === 'auto' ? userColorScheme : initialTheme || colorMode;
+
+  useEffect(() => {}, [colorMode, initialTheme]);
```

Comment:
Cruft?

Reply:
No it's needed to force React to re-render when the theme changes on localStorage. colorMode might not be doing much for us there tho (I use it in a refactor to move useColorScheme to a hook though)

Reply:
Hmm, I don't understand how that works. The only way for `initialTheme` to change is if this component is re-rendering, right?

Reply:
I would _think_ you'd want to listen for changes to localStorage for that case?

https://developer.mozilla.org/en-US/docs/Web/API/Window/storage_event